File: 5 - Matemática/1028.c
#include <stdio.h>

// Algoritmo de Euclides (Divisões Sucessivas)
// https://www.youtube.com/watch?v=SyhJFuw2u7c
int mdc(int x, int y) {
	int aux;

	do {
		aux = x % y;
		x = y;
		y = aux;
	} while (y > 1);
	return x;
}

int main() {
	int n, f1, f2;

	scanf("%d", &n);
	for (; n > 0; n--) {
		scanf("%d%d", &f1, &f2);

		if (f2 > f1) {
			printf("%d\n", mdc(f2, f1));
		} else {
			printf("%d\n", mdc(f1, f2));
		}
	}
	return 0;
}

File: 5 - Matemática/1028.py
# -*- coding: utf-8 -*-

for _ in range(int(input())):
	f1, f2 = [int(x) for x in input().strip().split()]

	# Se f2 > f1, troque os valores
	if f2 > f1:
		aux = f1
		f1 = f2
		f2 = aux

	# MDC por Divisões Sucessivas
	while f2 > 1:
		aux = f1 % f2;
		f1 = f2;
		f2 = aux;
	print(f1)

